### PR TITLE
script: Throw a `TypeError` when trying to create an `OffscreenCanvas` with an unknown context type

### PR DESCRIPTION
**Pull request by chickenleaf**
Previously, when an unknown context types was passed to the `getContext()` method of `OffscreenCanvas`, `null` was returned. This modifies the method to match the spec by returning `TypeError` in this case.

**Note:**
- [list of test failures to be fixed in offscreencanvas](https://github.com/servo/servo/tree/main/tests/wpt/meta/html/canvas/offscreen/canvas-context)
- except [this](https://github.com/servo/servo/blob/main/tests/wpt/meta/html/canvas/offscreen/canvas-context/2d.getContext.options.any.js.ini) others now pass.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] There are tests for these changes

### CODE REVIEW

File: components/script/dom/offscreencanvas.rs [hunk 1]
```diff
@@ -157,18 +157,27 @@ impl OffscreenCanvasMethods for OffscreenCanvas {
...
                 .map(OffscreenRenderingContext::WebGLRenderingContext),
             "webgl2" | "experimental-webgl2" => self
                 .get_or_init_webgl2_context(cx, options)
                 .map(OffscreenRenderingContext::WebGL2RenderingContext),*/
-            _ => None,
+            "2D" => Err(Error::Type(String::from(
+                "Context name \"2D\" is unrecognised; matching is case sensitive",
+            ))),
+            _ => Err(Error::Type(String::from("Unrecognized context identifier"))),
         }
```
Comment (mrobinson) [inline components/script/dom/offscreencanvas.rs:181]: I don't think you need a custom error message for each of these. Instead maybe just have a fallback case that return a generic error message like "Unrecognized OffscreenCanvas context type".

**Review by mrobinson — CHANGES_REQUESTED** [review]
Nice! I would just have a single fallback case and handle all unknown values there.

File: components/script/dom/offscreencanvas.rs (hunk 1 above)
Comment (chickenleaf) [reply components/script/dom/offscreencanvas.rs:181]: Alright!

**Review by mrobinson — APPROVED** [review]

**Comment by mrobinson** [comment]
This has caused some test failures and some new passes:

```
ERROR [expected OK] /_webgl/conformance/offscreencanvas/context-creation-worker.html
ERROR [expected OK] /_webgl/conformance/offscreencanvas/context-creation.html

    FAIL [expected PASS] subtest: WebGL test #0: successfullyParsed should be true (of type boolean). Was undefined (of type undefined).

    assert_true: successfullyParsed should be true (of type boolean). Was undefined (of type undefined). expected true got false

ERROR [expected OK] /_webgl/conformance/offscreencanvas/methods-worker.html
ERROR [expected OK] /_webgl/conformance/offscreencanvas/methods.html

    FAIL [expected PASS] subtest: WebGL test #0: successfullyParsed should be true (of type boolean). Was undefined (of type undefined).

    assert_true: successfullyParsed should be true (of type boolean). Was undefined (of type undefined). expected true got false

ERROR [expected OK] /_webgl/conformance2/offscreencanvas/context-creation.html

    FAIL [expected PASS] subtest: WebGL test #0: successfullyParsed should be true (of type boolean). Was undefined (of type undefined).

    assert_true: successfullyParsed should be true (of type boolean). Was undefined (of type undefined). expected true got false

ERROR [expected OK] /_webgl/conformance2/offscreencanvas/methods-2-worker.html
ERROR [expected OK] /_webgl/conformance2/offscreencanvas/methods-2.html

    FAIL [expected PASS] subtest: WebGL test #0: successfullyParsed should be true (of type boolean). Was undefined (of type undefined).

    assert_true: successfullyParsed should be true (of type boolean). Was undefined (of type undefined). expected true got false

OK /html/canvas/offscreen/canvas-context/2d.canvas.context.invalid.args.html

    PASS [expected FAIL] subtest: Calling getContext with invalid arguments.

OK /html/canvas/offscreen/manual/the-offscreen-canvas/offscreencanvas.getcontext.html

    PASS [expected FAIL] subtest: Test that getContext with un-supported string throws a TypeError.

OK /html/canvas/offscreen/manual/the-offscreen-canvas/offscreencanvas.getcontext.worker.html

    PASS [expected FAIL] subtest: Test that getContext with un-supported string throws a TypeError.
```

**Comment by jdm** [comment]
I'd like to understand the new errors in the webgl tests.

**Comment by jdm** [comment]
Oh, I bet it's because we don't support off screen webgl contexts yet, and previously we just returned null.

**Comment by chickenleaf** [comment]
> Oh, I bet it's because we don't support off screen webgl contexts yet, and previously we just returned null.

so i just need to update the expectations? since i replaced the return none, what do i do about the failed ones now?

**Comment by jdm** [comment]
> > Oh, I bet it's because we don't support off screen webgl contexts yet, and previously we just returned null.
> 
> so i just need to update the expectations? since i replaced the return none, what do i do about the failed ones now? do those tests need changes? like to expect a typeerror in some cases?

Updating the expectations is correct here, since ilthe test is using a feature that we don't support.

**Review by jdm — APPROVED** [review]

**Pull request merged**